feat(jobs): improve retries (#6703)
scaleway/docs-content: pages/serverless-jobs/how-to/manage-job-retries.mdx

import Requirements from '@macros/iam/requirements.mdx'

> The retry option lets you define how many times a Serverless Job automatically retries on failure. A retry triggers when the Job Run returns an [Exit Code](/serverless-jobs/concepts/#exit-code) different than `0`. Once the job reaches the maximum number of failed retries, it is permanently marked as `failed`.

Reviewer: suggestion: I think we need to explicit on which job run state we retry (`Failed`) and that we do not yet support retrying on `Interrupted` state for clarity. Also, mentioning non-zero exit code need to be removed because we may have a `Failed` job run with null exit code (example: platform mismatch) that will be retried.

```suggestion
The retry option lets you define how many times a Serverless Job automatically retries on failure. A retry is triggered when the job run falls in a `Failed` state. Retry on `Interrupted` state is not yet supported. Once the job reaches the maximum number of failed retries, it is permanently marked as `failed`.
```

> The retry option lets you define how many times a Serverless Job automatically retries on failure. A retry triggers when the job run reaches the `failed` [Status](/serverless-jobs/concepts/#status), the configured retry count is greater than zero, and the job run exits with a non-zero exit code. Once the job reaches the maximum number of failed retries, it is permanently marked as failed.

Reviewer 2: I don't get what is the added value for the user to go this deep in technical explanations ?
There is only one setting in the console, and it is named "Max retry attempts". So I would like to write the explanations around this and not the technical magic behind:

 - The user sets a maximum number of retry attempts.
 - A job run that does not succeed retries until the max retries number is reached.

> The retry option allows you to define how many times a Serverless Job automatically retries on failure. A retry is triggered when the job run reaches the `failed` [Status](/serverless-jobs/concepts/#status), the configured retry count is greater than zero, and the job run exits with a non-zero exit code. When the maximum number of failed retries is reached, the job is permanently marked as failed.

Reviewer 3: ```suggestion
The retry option lets you define how many times a Serverless Job automatically retries on failure. A retry triggers when the job run reaches the `failed` [Status](/serverless-jobs/concepts/#status), the configured retry count is greater than zero, and the job run exits with a non-zero exit code. Once the job reaches the maximum number of failed retries, it is permanently marked as failed.
```

Reviewer 2: I don't get what is the added value for the user to go this deep in technical explanations ?
